### PR TITLE
Fix handling of JSON data

### LINKED ISSUE #181: JSON data type not supported
Inserting data into a column with the MySQL JSON data type breaks the replication stream.

```
mysql> create table json_test (id int, value json);
mysql> insert into json_test (id, value) values (1, '{"my_key": "my_val"}');
Query OK, 1 row affected (0.01 sec)
```

Stack trace:
```
Traceback (most recent call last):
  File "/src/main.py", line 89, in <module>
    main()
  File "/src/main.py", line 72, in main
    for row in binlogevent.rows:
  File "/usr/lib/python2.7/site-packages/pymysqlreplication/row_event.py", line 416, in rows
    self._fetch_rows()
  File "/usr/lib/python2.7/site-packages/pymysqlreplication/row_event.py", line 411, in _fetch_rows
    self.__rows.append(self._fetch_one_row())
  File "/usr/lib/python2.7/site-packages/pymysqlreplication/row_event.py", line 464, in _fetch_one_row
    row["values"] = self._read_column_data(self.columns_present_bitmap)
  File "/usr/lib/python2.7/site-packages/pymysqlreplication/row_event.py", line 172, in _read_column_data
    (column.type))
NotImplementedError: Unknown MySQL column type: 245
```

my.cnf using MySQL 5.7 docker image `d9124e6c552f`:
```
[mysqld]
server-id        = 12345
log_bin          = /var/log/mysql/mysql-bin.log
expire_logs_days = 10
max_binlog_size  = 100M
binlog-format    = row
gtid_mode=ON
log-slave_updates=true
enforce_gtid_consistency
```

### PR DESCRIPTION
**Pull request by baloo**
Introduced in 5.7.8

I chose not to handle charset parsing, all dict returned are in binary form. User is expected to parse it.

Fixes #181

### CODE REVIEW

**Comment by baloo** [comment]
This is a very rough implementation that misses a LOT of types but this gives a good starting point. Do not merge it yet.

**Comment by baloo** [comment]
https://github.com/percona/percona-server/blob/5.7/sql/json_binary.cc Here is the official "documentation" for mysql json type

**Comment by baloo** [comment]
@namabile I'd love if you could give this a shot before I merge it.

**Comment by namabile** [comment]
Awesome. I'll pull this later today and give it a shot.

**Comment by namabile** [comment]
Gave this a try and looks really good.

I hit an error on json types 4 and 5. Here are the cases I tested:

`insert into test (id, value) values (1, 'true');`
`insert into test (id, value) values (1, '9');`

Let me know if I can provide any info to help debug.

**Comment by baloo** [comment]
Well, those are easy to fix though. I intentionally left those one out. I though json was spec'ed as either an object or an array. No other structure. I just re-read the spec. I was wrong.
I'll fix those.

File: pymysqlreplication/tests/test_data_type.py [hunk 1]
```diff
@@ -413,6 +425,94 @@ def test_geometry(self):
...
+            data = {'foo': t}
+            create_query = "CREATE TABLE test (id int, value json);"
+            insert_query = """INSERT INTO test (id, value) VALUES (1, '%s');""" % json.dumps(data)
+            event = self.create_and_insert_value(create_query, insert_query)
+            self.assertEqual(event.rows[0]["values"]["value"], to_binary_dict(data))
+
+            self.tearDown()
+            self.setUp()
+
+    def test_json_basic(self):
```
Comment (baloo) [inline pymysqlreplication/tests/test_data_type.py]: Here is a test for your last usecase @namabile

**Comment by baloo** [comment]
Hey @noplay, let me know what you think about this one.

**Pull request merged**